frontend/components/PaginatedList/PaginatedList.tsx

```diff
@@ -206,71 +209,76 @@ function PaginatedListInner<TItem extends Record<string, any>>(
           <Spinner />
         </div>
       )}
-      <div>
-        <ul className={`${baseClass}__list`}>
-          {heading && (
-            <li className={`${baseClass}__row ${baseClass}__header`}>
-              {heading}
+      <ul className={`${baseClass}__list`}>
+        {heading && (
+          <li className={`${baseClass}__row ${baseClass}__header`}>
+            {heading}
+          </li>
+        )}
+        {items.map((_item) => {
+          // If an item has been marked as changed, use the changed version
+          // of the item rather than the one from the page fetch.  This allows
+          // us to render an item correctly even after we've navigated away
+          // from its page and then back again.
+          const item = dirtyItems[_item[idKey]] ?? _item;
+
+          const rowClasses = classnames(`${baseClass}__row`, {
+            [`${baseClass}__row--disabled`]: item.disabled,
```

Comment:
I'm surprised Typescript lets us get away with this -- it has no idea if `disabled` is a valid property on an item.  By doing this we're sneaking business logic into this component (nothing too scandalous, but still).  A better option IMO would be to pass an `isItemDisabled` property to `PaginatedList`, like we do with `isSelected` (which I'm realizing now would be better named `isItemSelected`, but that's better fixed in a separate PR).

Reply:
great point